packages/application-shell/src/components/configure-intl-provider/configure-intl-provider.js

```diff
@@ -19,14 +19,16 @@ const hideAppLoader = () => {
 };
 
 const ConfigureIntlProvider = props => {
+  React.useEffect(() => {
+    if (props.locale) {
+      hideAppLoader();
+    }
+  }, [props.locale]);
```

Comment:
Curious: I thought this wouldn't work without a warning and would have to be `[locale]`?

Reply:
No idea, it works...or at least I didn't get any errors about it 🤔

Reply:
Interesting, I looked into this a bit more and it seems that the weird "destructuring" warning only appears if you call a prop function (see below):

```js
// all good
React.useEffect(() => {
  if (props.locale) {
    hideAppLoader();
  }
}, [props.locale]);

// error
React.useEffect(() => {
  if (props.locale) {
    hideAppLoader();
    props.onChange()
  }
}, [props.locale]);
```

Reply:
So by destructuring `onChange`, the warning is gone. It seems to me that this is more of a bug in the eslint plugin 🤔

Reply:
Ah, yeah then any assumption for reasoning I had doesn't make sense anymore 😄. Then can we just do something like:

```js
const { onChange } = props;

React.useEffect(() => {
  if (props.locale) {
    hideAppLoader();
    props.onChange()
  }
}, [props.locale, onChange]);
```

for now?

Reply:
Yes that's what I'm doing. We should maybe report this in the react repo